packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -171,6 +171,24 @@ export function matches(element: Element, selector: string | AST): boolean {
         pseudoClass.argument !== undefined &&
         matchCSSProperty(element, pseudoClass.argument, '::before')
       );
+    } else if (pseudoClass.name === 'xpath') {
+      const { argument } = pseudoClass;
+      if (argument === undefined) {
+        return false;
+      }
+
+      try {
+        // Create an XPath evaluator
+        const evaluator = new XPathEvaluator();
+        // Create an XPath expression
+        const expression = evaluator.createExpression(argument, null);
+        // Evaluate the expression
+        const result = expression.evaluate(element, XPathResult.BOOLEAN_TYPE, null);
+        return result.booleanValue;
```

Comment:
This `XPathEvaluation` must be stored and cached somewhere to avoid bottleneck. Also, consider not to use it or find any other alternatives for `XPathEvaluator` as W3C DOM API specifies it as heavily outdated: https://github.com/whatwg/dom/issues/67. Also, see https://dom.spec.whatwg.org/#xpath.

> DOM Level 3 XPath defined an API for evaluating XPath 1.0 expressions. These APIs are widely implemented, but have not been maintained. The interface definitions are maintained here so that they can be updated when Web IDL changes. Complete definitions of these APIs remain necessary and such work is tracked and can be contributed to in whatwg/dom#67.